### DISCUSSION TITLE
debug  with shared lib,not find shared file

### DISCUSSION THREAD

**Discussion by youngday** [xmake-io/xmake#3675] [Q&A]
### Xmake Version

2.7.8

### Operating System Version and Architecture

ubuntu20.04

### Describe Bug

    for linux,linkdirs,comment below,

    target("test")
        add_links("testso")   
        add_linkdirs("$(buildir)/lib") --just for shared lib link path
        add_rpathdirs("$(buildir)/lib") --just for shared lib link path when running on linux,but not worked
        add_rpathdirs("@loader_path/lib") --just for shared lib link path when running on linux,worked well

 

### Expected Behavior

easily link and find shared lib when  compile and debug at time.

**Comment by waruqi** [comment]
`$(buildir)` 获取的是相对路径 build，不是绝对路径，不建议使用它设置到 rpath，走 `@loader_path` 设置，具体看文档
